src/shared/clients/defaultStsClient.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { STS } from 'aws-sdk'
+import { ServiceConfigurationOptions } from 'aws-sdk/lib/service'
+import { ext } from '../extensionGlobals'
+import { StsClient } from './stsClient'
+
+export class DefaultStsClient implements StsClient {
+
+    public constructor (
+        public readonly regionCode: string,
+        private readonly _credentials?: ServiceConfigurationOptions
```

Comment:
Nit: Re offline discussion, we don't use `_` prefixes except where disambiguation is required.